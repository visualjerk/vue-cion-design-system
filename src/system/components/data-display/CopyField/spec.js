import { shallowMount } from '@vue/test-utils'
import Comp from './CopyField.vue'

describe('CopyField.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Comp)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('defaults to div', () => {
    const wrapper = shallowMount(Comp)
    expect(wrapper.props().tag).toEqual('div')
  })

  it('displays text', () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        default: 'Test',
      },
    })
    expect(wrapper.text()).toEqual('Test')
  })
})
