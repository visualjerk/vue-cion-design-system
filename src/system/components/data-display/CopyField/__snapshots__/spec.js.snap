// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyField.vue matches snapshot 1`] = `
<div
  class="ds-copy-field ds-copy-field-base"
>
  <div />
   
  <div
    class="ds-copy-field-link"
  >
    <ds-button-stub
      color="soft"
      ghost="true"
      icon="copy"
      linktag="button"
    />
  </div>
   
  <transition-stub
    name="ds-copy-field-message"
  >
    <div
      class="ds-copy-field-message"
      style="display: none;"
    >
      <div
        class="ds-copy-field-message-text"
      />
    </div>
  </transition-stub>
</div>
`;
